File: packages/watchmen-model/src/watchmen_model/dqc/catalog.py
from typing import List, Optional, TypeVar

from watchmen_utilities import ExtendedBaseModel

from watchmen_model.common import OptimisticLock, TenantBasedTuple, TopicId, UserId

CatalogId = TypeVar('CatalogId', bound=str)


class Catalog(TenantBasedTuple, OptimisticLock, ExtendedBaseModel):
	catalogId: CatalogId = None
	name: str = None
	topicIds: Optional[List[TopicId]] = []
	techOwnerId: UserId = None
	bizOwnerId: UserId = None
	tags: Optional[List[str]] = []
	description: str = None


class CatalogCriteria(ExtendedBaseModel):
	name: Optional[str] = None
	topicId: Optional[TopicId] = None
	techOwnerId: Optional[UserId] = None
	bizOwnerId: Optional[UserId] = None

File: packages/watchmen-dqc-client/src/models/catalog.ts
// Catalog models — mirror watchmen-model dqc/catalog.py.
// Source: packages/watchmen-model/src/watchmen_model/dqc/catalog.py
// Endpoints: GET/POST/DELETE /dqc/catalog, POST /dqc/catalog/criteria
// (packages/watchmen-rest-dqc/.../admin/catalog_router.py)

/** `Catalog` model (tenant-scoped tuple with optimistic lock). */
export interface Catalog {
	catalogId?: string;
	name?: string;
	topicIds?: string[];
	techOwnerId?: string;
	bizOwnerId?: string;
	tags?: string[];
	description?: string;
	tenantId?: string;
	version?: number;
}

/** `CatalogCriteria` model — search criteria for POST /dqc/catalog/criteria. */
export interface CatalogCriteria {
	name?: string;
	topicId?: string;
	techOwnerId?: string;
	bizOwnerId?: string;
}
